api/src/controllers/websocket_rpc.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+module.exports = WebsocketRpcControllerFactory
+
+const ws = require('ws')
+const debug = require('debug')('ilp-kit:websocket-rpc')
+const Connector = require('../lib/connector')
+
+function WebsocketRpcControllerFactory (deps) {
+  const connector = deps(Connector)
+
+  return class WebsocketRpcController {
+    static init (server) {
+     const socketServer = new ws.Server({
+        perMessageDeflate: false,
+        server
+      })
+
+      debug('listening for websocket connections')
+      socketServer.on('connection', socket => {
+        const url = socket.upgradeReq.url
+        const params = url.match(/^\/api\/peers\/rpc\/ws\/(.+?)\/(.+?)\/?$/)
+        if (!params) {
+          debug('Error: no ws at "' + url + '"')
+          socket.close()
+          return
+        }
+
+        const [ , prefix, token ] = params
+        let plugin
+
+        try {
+          plugin = connector.getPlugin(prefix)
+        } catch (e) {
+          debug(e)
+          socket.close()
+          return
+        }
+
+        if (!plugin || !plugin.isAuthorized(token)) {
+          debug(`Error: prefix (${prefix}) and/or token (${token}) are incorrect`)
+          socket.close()
+          return
+        }
+        
+        if (typeof plugin.addSocket === 'function') {
+          plugin.addSocket(socket)
```

Comment:
should we check that no socket is connected to it yet? or do you want to support multiple connected clients?

Reply:
I think we should support multiple connected clients, especially if we plan on one of these plugins being an admin plugin containing all the clients at once